runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsPlugin.java

```diff
@@ -47,10 +59,63 @@
 	@Inject
 	private OverlayManager overlayManager;
 
+	@Inject
+	private Client client;
+
+	@Inject
+	private StatusBarsConfig config;
+
+	@Getter(AccessLevel.PACKAGE)
+	private Instant lastCombatAction;
+
 	@Override
 	protected void startUp() throws Exception
 	{
-		overlayManager.add(overlay);
+	}
+
+	void updateLastCombatAction()
+	{
+		this.lastCombatAction = Instant.now();
+	}
+
+	@Subscribe
+	public void onGameTick(GameTick gameTick)
+	{
+		if (!config.toggleRestorationBars())
+		{
+			overlayManager.add(overlay);
+			return;
+		}
+		else
+		{
+			hideStatusBar();
+		}
+	}
+
+	private void hideStatusBar()
+	{
+		final Actor interacting = client.getLocalPlayer().getInteracting();
+		final boolean isNpc = interacting instanceof NPC;
+		final int combatTimeout = config.hideStatusBarDelay();
+
+		if (isNpc)
+		{
+			final NPC npc = (NPC) interacting;
+			final NPCComposition npcComposition = npc.getComposition();
+			final List<String> npcMenuActions = Arrays.asList(npcComposition.getActions());
+			if (npcMenuActions.contains("Attack") && config.toggleRestorationBars())
+			{
+				updateLastCombatAction();
+				overlayManager.add(overlay);
+			}
+		}
+		else if (lastCombatAction != null)
+		{
+			if (Duration.between(getLastCombatAction(), Instant.now()).getSeconds() > combatTimeout)
```

Comment:
`combatTimeout` is used exactly once, move its contents into this condition.

Also, the `else if (lastCombatAction != null)` block contains only this if statement, combine these conditions.